File: old/tags/2.1/cpp/DIS/IntercomCommunicationsParameters.cpp
#include <DIS/IntercomCommunicationsParameters.h> 

using namespace DIS;


IntercomCommunicationsParameters::IntercomCommunicationsParameters():
   _recordType(0), 
   _recordLength(0), 
   _recordSpecificField(0)
{
}

IntercomCommunicationsParameters::~IntercomCommunicationsParameters()
{
}

unsigned short IntercomCommunicationsParameters::getRecordType() const
{
    return _recordType;
}

void IntercomCommunicationsParameters::setRecordType(unsigned short pX)
{
    _recordType = pX;
}

unsigned short IntercomCommunicationsParameters::getRecordLength() const
{
    return _recordLength;
}

void IntercomCommunicationsParameters::setRecordLength(unsigned short pX)
{
    _recordLength = pX;
}

unsigned int IntercomCommunicationsParameters::getRecordSpecificField() const
{
    return _recordSpecificField;
}

void IntercomCommunicationsParameters::setRecordSpecificField(unsigned int pX)
{
    _recordSpecificField = pX;
}

void IntercomCommunicationsParameters::marshal(DataStream& dataStream) const
{
    dataStream << _recordType;
    dataStream << _recordLength;
    dataStream << _recordSpecificField;
}

void IntercomCommunicationsParameters::unmarshal(DataStream& dataStream)
{
    dataStream >> _recordType;
    dataStream >> _recordLength;
    dataStream >> _recordSpecificField;
}


bool IntercomCommunicationsParameters::operator ==(const IntercomCommunicationsParameters& rhs) const
 {
     bool ivarsEqual = true;

     if( ! (_recordType == rhs._recordType) ) ivarsEqual = false;
     if( ! (_recordLength == rhs._recordLength) ) ivarsEqual = false;
     if( ! (_recordSpecificField == rhs._recordSpecificField) ) ivarsEqual = false;

    return ivarsEqual;
 }

int IntercomCommunicationsParameters::getMarshalledSize() const
{
   int marshalSize = 0;

   marshalSize = marshalSize + 2;  // _recordType
   marshalSize = marshalSize + 2;  // _recordLength
   marshalSize = marshalSize + 4;  // _recordSpecificField
    return marshalSize;
}

File: old/tags/2.2/java/edu/nps/moves/dis/IntercomCommunicationsParameters.java
package edu.nps.moves.dis;

import java.util.*;
import java.io.*;
import edu.nps.moves.jaxb.dis.*;

/**
 * 5.2.46.  Intercom communcations parameters
 *
 * Copyright (c) 2007, MOVES Institute, Naval Postgraduate School. All rights reserved.
 * This work is licensed under the BSD open source license, available at https://www.movesinstitute.org/licenses/bsd.html
 *
 * @author DMcG
 */
public class IntercomCommunicationsParameters extends Object implements Serializable
{
   /** Type of intercom parameters record */
   protected int  recordType;

   /** length of record */
   protected int  recordLength;

   /** Jerks. Looks like the committee is forcing a lookup of the record type parameter to find out how long the field is. This is a placeholder. */
   protected long  recordSpecificField;


/** Constructor */
 public IntercomCommunicationsParameters()
 {
 }

/** 
 * Constructor--takes a parallel jaxb object and returns an open-dis object 
 * 1.4_sed_bait_start */
 public IntercomCommunicationsParameters(edu.nps.moves.jaxb.dis.IntercomCommunicationsParameters x)
 {
     this.recordType = x.getRecordType();
     this.recordLength = x.getRecordLength();
     this.recordSpecificField = x.getRecordSpecificField();
 }
/* 1.4_sed_bait_end */


/**
 * returns a jaxb object intialized from this object, given an empty jaxb object
 * 1.4_sed_bait_start **/
 public edu.nps.moves.jaxb.dis.IntercomCommunicationsParameters initializeJaxbObject(edu.nps.moves.jaxb.dis.IntercomCommunicationsParameters x)
 {
     ObjectFactory factory = new ObjectFactory();

     x.setRecordType( this.getRecordType() );
     x.setRecordLength( this.getRecordLength() );
     x.setRecordSpecificField( this.getRecordSpecificField() );
   return x;
 }
/* 1.4_sed_bait_end */


public int getMarshalledSize()
{
   int marshalSize = 0; 

   marshalSize = marshalSize + 2;  // recordType
   marshalSize = marshalSize + 2;  // recordLength
   marshalSize = marshalSize + 4;  // recordSpecificField

   return marshalSize;
}


public void setRecordType(int pRecordType)
{ recordType = pRecordType;
}

public int getRecordType()
{ return recordType; 
}

public void setRecordLength(int pRecordLength)
{ recordLength = pRecordLength;
}

public int getRecordLength()
{ return recordLength; 
}

public void setRecordSpecificField(long pRecordSpecificField)
{ recordSpecificField = pRecordSpecificField;
}

public long getRecordSpecificField()
{ return recordSpecificField; 
}


public void marshal(DataOutputStream dos)
{
    try 
    {
       dos.writeShort( (short)recordType);
       dos.writeShort( (short)recordLength);
       dos.writeInt( (int)recordSpecificField);
    } // end try 
    catch(Exception e)
    { 
      System.out.println(e);}
    } // end of marshal method

public void unmarshal(DataInputStream dis)
{
    try 
    {
       recordType = dis.readShort();
       recordLength = dis.readShort();
       recordSpecificField = dis.readInt();
    } // end try 
   catch(Exception e)
    { 
      System.out.println(e); 
    }
 } // end of unmarshal method 


 /**
  * The equals method doesn't always work--mostly on on classes that consist only of primitives. Be careful.
  */
 public boolean equals(IntercomCommunicationsParameters rhs)
 {
     boolean ivarsEqual = true;

    if(rhs.getClass() != this.getClass())
        return false;

     if( ! (recordType == rhs.recordType)) ivarsEqual = false;
     if( ! (recordLength == rhs.recordLength)) ivarsEqual = false;
     if( ! (recordSpecificField == rhs.recordSpecificField)) ivarsEqual = false;

    return ivarsEqual;
 }
} // end of class
